src/mistral_common/experimental/app/main.py

```diff
@@ -0,0 +1,279 @@
+    """
+
+    app_name: str = "Mistral-common API"
+    app_version: str = importlib.metadata.version("mistral-common")
+
+    def model_post_init(self, context: Any) -> None:
+        super().model_post_init(context)
+        self._tokenizer: Optional[MistralTokenizer] = None
+
+    def _load_tokenizer(self, tokenizer_path: Union[str, Path], validation_mode: ValidationMode) -> None:
+        if tokenizer_path == "":
+            raise ValueError("Tokenizer path must be set via the environment variable `TOKENIZER_PATH`.")
+        elif self._tokenizer is not None:
+            raise ValueError("Tokenizer has already been initialized.")
+
+        if isinstance(tokenizer_path, str):
+            candidate_tokenizer_path = Path(tokenizer_path)
+            if candidate_tokenizer_path.exists():
+                tokenizer_path = candidate_tokenizer_path
+
+        if isinstance(tokenizer_path, Path) and tokenizer_path.exists():
+            self._tokenizer = MistralTokenizer.from_file(tokenizer_path, mode=validation_mode)
+        else:
+            self._tokenizer = MistralTokenizer.from_hf_hub(str(tokenizer_path), mode=validation_mode)
+
+    @property
+    def tokenizer(self) -> MistralTokenizer:
+        if self._tokenizer is None:
+            raise ValueError("Tokenizer not initialized.")
+        return self._tokenizer
+
+    @tokenizer.setter
+    def tokenizer(self, value: MistralTokenizer) -> None:
+        if not isinstance(value, MistralTokenizer):
+            raise ValueError("Tokenizer must be an instance of MistralTokenizer.")
+        self._tokenizer = value
+
+
+main_router = APIRouter(tags=["app"])
+tokenize_router = APIRouter(prefix="/tokenize", tags=["tokenizer", "tokenize"])
+decode_router = APIRouter(prefix="/detokenize", tags=["tokenizer", "detokenize"])
+
+
+def get_settings() -> Settings:
+    r"""Get the settings for the Mistral-common API."""
+    return Settings()
+
+
+@main_router.get("/")
+def redirect_to_docs() -> RedirectResponse:
+    r"""Redirect to the documentation."""
+    return RedirectResponse(url="docs")
+
+
+@tokenize_router.post("/request")
+def tokenize_request(
+    request: Union[ChatCompletionRequest, OpenAIChatCompletionRequest],
+    settings: Annotated[Settings, Depends(get_settings)],
+) -> list[int]:
+    r"""Tokenize a chat completion request."""
+    if isinstance(request, OpenAIChatCompletionRequest):
+        try:
+            request = ChatCompletionRequest.from_openai(**request.model_dump(exclude_none=True))
+        except (ValidationError, ValueError) as e:
+            raise HTTPException(status_code=400, detail=str(e))
+
+    if request.messages == []:
+        raise HTTPException(status_code=400, detail="Messages list cannot be empty.")
+
+    tokenized = settings.tokenizer.encode_chat_completion(request)
+    assert isinstance(tokenized, Tokenized), type(tokenized)
+    return tokenized.tokens
+
+
+@decode_router.post("/string")
+def detokenize_to_string(
```

Comment:
I wouldn't keep this but ok if you feel strongly

Reply:
See above, let's keep it for now.